server/src/main/java/org/elasticsearch/index/codec/tsdb/es819/XDocValuesConsumer.java

```diff
@@ -152,6 +153,102 @@ public long longValue() throws IOException {
         };
     }
 
+    /** Tracks state of one binary sub-reader that we are merging */
+    private static class BinaryDocValuesSub extends DocIDMerger.Sub {
```

Comment:
This is copied from Lucene's `DocValuesConsumer`?

Reply:
Yes, except that it returns an anonymous subclass of `TsdbDocValuesProducer` instead of `EmptyDocValuesProducer` so that it can support merge stats.